packages/admin/admin/src/hooks/useAsyncOptionsProps.ts

```diff
@@ -11,12 +11,15 @@ export interface AsyncOptionsProps<T> {
 export function useAsyncOptionsProps<T>(loadOptions: () => Promise<T[]>): AsyncOptionsProps<T> {
     const [open, setOpen] = useState(false);
     const [options, setOptions] = useState<T[]>([]);
-    const loading = open && options.length === 0;
+    const [loading, setLoading] = useState(false);
 
     const handleOpen = async () => {
         setOpen(true);
+        setLoading(true);
         setOptions([]); // Reset options to show loading
```

Comment:
now with the explicit loading state, this comment is not valid anymore and should be removed. Maybe the whole `setOptions([]);` is not needed anymore?

Reply:
removed the comment and removed setOptions in this hook.

I also added an additional loading handling in the `FinalFormSelect` -> not rendering options when the hook is currently loading.